### PR TITLE
feat: expose CanonizerUtil

### PR DESCRIPTION
**Pull request by lpatiny**

### CODE REVIEW

**Comment by lpatiny** [comment]
@targos I'm trying to expose the static method CanonizerUtil.getIDCode() however in the build CanonizerUtil does not appear and I don't know why.

Where should I specify that this class is publicly available ?

**Comment by targos** [comment]
If it doesn't appear in the build, that means there is an error somewhere. You can pass the `--verbose` flag to the build script and it will create a log file.

File: src/com/actelion/research/gwt/core/JSCanonizerUtil.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+public class JSCanonizerUtil {
+
+	public static int NORMAL = CanonizerUtil.NORMAL;
+	public static int NOSTEREO = CanonizerUtil.NOSTEREO;
+	public static int BACKBONE = CanonizerUtil.BACKBONE;
+	public static int TAUTOMER = CanonizerUtil.TAUTOMER;
+	public static int NOSTEREO_TAUTOMER = CanonizerUtil.NOSTEREO_TAUTOMER;
+
+	public static String getIDCode(JSMolecule mol, IDCODE_TYPE type) {
+		return CanonizerUtil.getIDCode(mol, false);
```
Comment (targos) [inline src/com/actelion/research/gwt/core/JSCanonizerUtil.java:17]: This at least seems wrong. I don't think `CanonizerUtil` accepts a `JSMolecule`. If it accepts a StereoMolecule, you can call `mol.getStereoMolecule()`.

File: src/com/actelion/research/gwt/core/JSCanonizerUtil.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@JsType(name = "CanonizerUtil")
+public class JSCanonizerUtil {
+
+	public static int NORMAL = CanonizerUtil.NORMAL;
+	public static int NOSTEREO = CanonizerUtil.NOSTEREO;
+	public static int BACKBONE = CanonizerUtil.BACKBONE;
+	public static int TAUTOMER = CanonizerUtil.TAUTOMER;
+	public static int NOSTEREO_TAUTOMER = CanonizerUtil.NOSTEREO_TAUTOMER;
+
+	public static String getIDCode(JSMolecule mol, IDCODE_TYPE type) {
```
Comment (targos) [inline src/com/actelion/research/gwt/core/JSCanonizerUtil.java:16]: What is `IDCODE_TYPE`? Not sure how it will work in JS.

File: README.md [hunk 3]
```diff
@@ -35,6 +35,7 @@ It contains a structure viewer and a structure editor for browser applications.
 - [Molecule](https://cheminfo.github.io/openchemlib-js/classes/Molecule.html)
 - [Reaction](https://cheminfo.github.io/openchemlib-js/classes/Reaction.html)
 - [RingCollection](https://cheminfo.github.io/openchemlib-js/classes/RingCollection.html)
+- [CanonizerUtil](https://cheminfo.github.io/openchemlib-js/classes/CanonizerUtil.html)
```
Comment (targos) [inline README.md:38]: Please keep alphabetical order.

File: __tests__/library.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ describe('Checking for the presence of main APIs', () => {
     'RingCollection',
     'SDFileParser',
     'SSSearcher',
+    'CanonizerUtil',
```
Comment (targos) [inline __tests__/library.js:14]: You put `JSCanonizerUtil.java` in the "core" folder, so this should be part of `coreAPI`.

File: minimal.d.ts [hunk 5]
```diff
@@ -4,12 +4,13 @@ export {
   IHoseCodesOptions,
   Molecule,
   MolecularFormula,
+  CanonizerUtil,
```
Comment (targos) [inline minimal.d.ts:7]: same

File: README.md [hunk 6]
```diff
@@ -32,6 +32,7 @@ It contains a structure viewer and a structure editor for browser applications.
 
 ### Modules present in minimal, core and full builds
 
+- [CanonizerUtil](https://cheminfo.github.io/openchemlib-js/classes/CanonizerUtil.html)
```
Comment (targos) [inline README.md:35]: this should be in the next section

File: minimal.d.ts [hunk 7]
```diff
@@ -1,4 +1,5 @@
 export {
+  CanonizerUtil,
```
Comment (targos) [inline minimal.d.ts:2]: this should be in `core.d.ts`

**Pull request merged**